### PR TITLE
fix(op_crates/web) fix window constructor length

### PR DESCRIPTION
**Pull request by benjamingr**
Another small PR for web compatibility.

### CODE REVIEW

File: op_crates/web/03_global_interfaces.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
   const illegalConstructorKey = Symbol("illegalConstuctorKey");
 
   class Window extends EventTarget {
-    constructor(key) {
+    constructor(key = null) {
       if (key !== illegalConstructorKey) {
         throw new TypeError("Illegal constructor.");
       }
```
Comment (bartlomieju) [inline op_crates/web/03_global_interfaces.js:10]: We got a couple more such constructors, @benjamingr do you want to check all of them in a single PR?

Comment (benjamingr) [reply op_crates/web/03_global_interfaces.js:10]: Yes if a single PR is easier to review I'll push all the changes in a single PR :]

Comment (benjamingr) [reply op_crates/web/03_global_interfaces.js:10]: I've put all the changes here (in this PR) and verified in the spec that this fix is indeed correct (and required) for it.

**Review by bartlomieju — APPROVED** [review]
I've confirmed that this is inline with the browser. LGTM, thanks @benjamingr!

**Comment by benjamingr** [comment]
@bartlomieju for future reference to my understanding this is defined by the  [WebIDL spec](https://heycam.github.io/webidl/#ecmascript-binding) under "3. ECMAScript binding" :]

For example for AbortSignal - it is defined in the [DOM spec here](https://dom.spec.whatwg.org/#interface-AbortSignal), its behavior as a WebIDL interface [is defined here](https://heycam.github.io/webidl/#interface-object) and specifically step 10:

> Let length be 0.

**Pull request merged**